kong/router.lua

```diff
@@ -1739,12 +1739,15 @@ function _M.new(routes)
     end
 
   else -- stream
-    function self.exec(ctx)
+    local server_name = require("ngx.ssl").server_name
+
+    function self.exec()
       local src_ip = var.remote_addr
       local src_port = tonumber(var.remote_port, 10)
       local dst_ip = var.server_addr
       local dst_port = tonumber(var.server_port, 10)
-      local sni = ctx.sni_server_name
+      -- error value for non-TLS connections ignored intentionally
+      local sni, _ = server_name()
```

Comment:
@dndx for my own knowledge, is this at all different from var.ssl_server_name?

Reply:
This is faster, because the preread module doesn't have to run.

Reply:
It of course not a blocker, but in this case could we update the HTTP subsystem branch to use the faster alternative?